### ISSUE TITLE
Null value is displayed as "java.lang.Object@XXXXXX"

### ISSUE THREAD

**Issue by ApplePedlar** [jknack/handlebars.java#458]
I found bug of [Handlebars version 4.0.1](http://search.maven.org/#artifactdetails%7Ccom.github.jknack%7Chandlebars%7C4.0.1%7Cjar).

```java
package misc;

import java.io.IOException;

import com.github.jknack.handlebars.Handlebars;

public class Handlebars4Test {

    public static void main(String[] args) throws IOException {
        Handlebars handlebars = new Handlebars();
        Foo foo1 = new Foo("baa");
        Foo foo2 = new Foo(null);
        System.out.println("foo1: " + handlebars.compileInline("{{baa}}").apply(foo1));
        System.out.println("foo1 using \"this\": " + handlebars.compileInline("{{this.baa}}").apply(foo1));
        System.out.println("foo1 using \"with\": " + handlebars.compileInline("{{#with this}}{{baa}}{{/with}}").apply(foo1));
        System.out.println("foo2: " + handlebars.compileInline("{{baa}}").apply(foo2));
        System.out.println("foo2 using \"this\": " + handlebars.compileInline("{{this.baa}}").apply(foo2));
        System.out.println("foo2 using \"with\": " + handlebars.compileInline("{{#with this}}{{baa}}{{/with}}").apply(foo2));
    }

    static class Foo  {
        private String baa;
        
        public Foo(String baa) {
            this.baa = baa;
        }
        public String getBaa() {
            return baa;
        }
    }
}
```

result
```
foo1: baa
foo1 using "this": baa
foo1 using "with": baa
foo2: 
foo2 using "this": java.lang.Object@256216b3
foo2 using "with": java.lang.Object@256216b3
```

Thank you.

**Comment by ApplePedlar** [comment]
Handlebars version 4.0.0 result

```
foo1: baa
foo1 using "this": baa
foo1 using "with": baa
foo2: 
foo2 using "this": 
foo2 using "with": 
```

**Comment by jknack** [comment]
Fixed in 4.0.2.

Thanks
